ui/app/adapters/task-state.js

```diff
@@ -0,0 +1,37 @@
+import ApplicationAdapter from './application';
+import { inject as service } from '@ember/service';
+
+export default ApplicationAdapter.extend({
+  token: service(),
+
+  ls(model, path) {
+    return this.token
+      .authorizedRequest(`/v1/client/fs/ls/${model.allocation.id}?path=${path}`)
+      .then(handleFSResponse);
+  },
+
+  stat(model, path) {
+    return this.token
+      .authorizedRequest(`/v1/client/fs/stat/${model.allocation.id}?path=${path}`)
+      .then(handleFSResponse);
+  },
+});
+
+async function handleFSResponse(response) {
+  if (response.ok) {
+    return response.json();
+  } else {
+    const body = await response.text();
+
+    // FIXME is this the same across all platforms?
+    const statusIs500 = response.status === 500;
+    const bodyIncludes404Text = body.includes('no such file or directory');
+
+    const translatedCode = statusIs500 && bodyIncludes404Text ? 404 : response.status;
```

Comment:
I’m awaiting feedback on whether 500-as-404 works similarly across platforms, so this may not be quite ready, but everything else is apart from the custom Mirage scenario that I can remove later.

Reply:
I changed this to a `TODO` as I think it’s fine to leave in for now and return to if the API changes.